### PR TITLE
Fixing misspells reported by goreportcard

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
Fixing misspells as reported by https://goreportcard.com/report/github.com/aws/amazon-ecs-agent#misspell

### Implementation details
Misspells are mostly within comments, and few error log entries.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: N/A
Note: functional tests on Linux is likely to be failing due to flakey test, unable to find more information from the run log.

### Description for the changelog

Didn't update changelog since this isn't a bug fix or implementation of a new feature. 

### Licensing

This contribution is under the terms of the Apache 2.0 License:

### CODE REVIEW

File: agent/ecs_client/model/ecs/api.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ const opCreateCluster = "CreateCluster"
 
 // CreateClusterRequest generates a "aws/request.Request" representing the
 // client's request for the CreateCluster operation. The "output" return
-// value will be populated with the request's response once the request complets
-// successfuly.
+// value will be populated with the request's response once the request completes
```
Comment (samuelkarp) [inline agent/ecs_client/model/ecs/api.go]: Please revert changes to this file.  This probably means we need to regenerate the JSON model from the proper doc model, and then regenerate the client here.

File: agent/engine/task_manager.go [hunk 2]
```diff
@@ -558,7 +558,7 @@ type containerTransitionFunc func(container *api.Container, nextStatus api.Conta
 // It returns a transition struct including the information:
 // * container state it should transition to,
 // * a bool indicating whether any action is required
-// * an error indicate why this transition can't happend
+// * an error indicate why this transition couldn't happen
```
Comment (samuelkarp) [inline agent/engine/task_manager.go]: nit: I prefer "can't" here, though I'd also change it to "indicating".

File: agent/ecs_client/model/ecs/api.go (hunk 1 above)
Comment (aaithal) [reply agent/ecs_client/model/ecs/api.go]: +1. both this and https://github.com/aws/amazon-ecs-agent/pull/1156/files#diff-857ccf24765dde4c0cde56a3e67394fc are auto-generated. Can you please revert both of these files?

**Comment by yunhee-l** [comment]
Reverted spelling fixes in auto-generated files, and updated grammar in agent/engine/task_manager.go comment.

**Pull request merged**